### PR TITLE
fix(table): bubble up click event when row item is clicked

### LINKED ISSUE #944: Need to bubble up rowClick event on the click of table row
For  https://angular.carbondesignsystem.com/?path=/story/table--with-toolbar ,  
   need to bubble up separate `rowClick` event on the table row click without changing any configuration to toggle between checkbox select and row select 

## Detailed description

> Is this a feature request (new component, new icon), a bug, or a general issue?
feature request

> Is this issue related to a specific component?
https://angular.carbondesignsystem.com/?path=/story/table--with-toolbar
https://github.com/IBM/carbon-components-angular/blob/master/src/table/table.component.ts

> What did you expect to happen? What happened instead? What would you like to see changed?
Expected : Table  should be like gmail inbox where clicking on the row opens up the email and clicking on the checkbox selects it.  Should be able to do both simultaneously
Happened :  `singleSelect` enabled and `showSelectionColumn` disabled ,   `selectRow` is emitted when the row is clicked.  checkbox select and row select should not be mutually exclusive.
Requirement : Need to be able to select checkboxes as well as rows with the same config.

> What browser are you working in?
Chrome

> What version of the Carbon Design System are you using?
carbon-components-angular@3.28.0

> What offering/product do you work on? Any pressing ship or release dates we should be aware of?
ECAM.

### PR DESCRIPTION
**Pull request by Donisius**
Closes #944

### CODE REVIEW

File: src/table/body/table-row.component.ts [hunk 1]
```diff
@@ -151,6 +155,13 @@ export class TableRowComponent {
 	 */
 	@Output() expandRow = new EventEmitter();
 
+	/**
+	 * Emits when a row is not dependant on enableSingleSelect or showSelectionColumn.
+	 * SHould only get emitted when a row item is selected excluding expandButtons,
```
Comment (zvonimirfras) [inline src/table/body/table-row.component.ts]:
```suggestion
	 * Should only get emitted when a row item is selected excluding expand buttons,
```

File: src/table/body/table-row.component.ts [hunk 2]
```diff
@@ -151,6 +155,13 @@ export class TableRowComponent {
 	 */
 	@Output() expandRow = new EventEmitter();
 
+	/**
+	 * Emits when a row is not dependant on enableSingleSelect or showSelectionColumn.
```
Comment (zvonimirfras) [inline src/table/body/table-row.component.ts]:
```suggestion
	 * Emits when a row is not dependant on `enableSingleSelect` or `showSelectionColumn`.
```

File: src/table/body/table-row.component.ts [hunk 3]
```diff
@@ -59,7 +59,9 @@ import { Observable } from "rxjs";
 					[ngStyle]="model.getHeader(j).style"
 					[skeleton]="skeleton"
 					[attr.colspan]="item.colSpan"
-					[attr.rowspan]="item.rowSpan">
+					[attr.rowspan]="item.rowSpan"
+					(click)="onRowClick()"
```
Comment (zvonimirfras) [inline src/table/body/table-row.component.ts]: So adding these once on `tr` instead of every `td` doesn't work?

Comment (Donisius) [reply src/table/body/table-row.component.ts]: Adding them once on `tr` would also get triggered if a selection/expansion column was clicked

File: src/table/body/table-row.component.ts [hunk 4]
```diff
@@ -151,6 +155,13 @@ export class TableRowComponent {
 	 */
 	@Output() expandRow = new EventEmitter();
 
+	/**
+	 * Emits when a row is not dependant on `enableSingleSelect` or `showSelectionColumn`.
```
Comment (cal-smith) [inline src/table/body/table-row.component.ts]:
```suggestion
	 * Emits when a row is clicked regardless of `enableSingleSelect` or `showSelectionColumn`.
```

**Comment by sparikh1594** [comment]
Also waiting for this to be delivered

**Pull request merged**